File: 1746-largest-substring-between-two-equal-characters/largest-substring-between-two-equal-characters.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int ans = -1;
        List<Integer>[] map = new List[26];
        for(int i=0; i<s.length(); i++) {
            char ch = s.charAt(i);
            if(map[ch-'a'] == null) map[ch-'a'] = new ArrayList<>();
            map[ch-'a'].add(i);
        }
        for(int i=0; i<26; i++) {
            if(map[i] != null && map[i].size() >= 2) {
                ans = Math.max(ans, map[i].get(map[i].size()-1) - map[i].get(0) - 1);
            }
        }
        return ans;
    }
}

File: 1746-largest-substring-between-two-equal-characters/largest-substring-between-two-equal-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var maxLengthBetweenEqualCharacters = function(s) {
    let map = new Map();
    let ans = -1;
    for(let i=0; i<s.length; i++) {
        if(!map.has(s[i])) map.set(s[i], i);
        else ans = Math.max(ans, i - map.get(s[i]) - 1);
    }
    return ans;
};